### PR TITLE
feat(notifications): emphasize notification's title

### PR DESCRIPTION
**Pull request by mareklibra**
## Hey, I just made a Pull Request!

The notification's title is emphasized to be clearly distinguished from the description.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mareklibra** [comment]
emphasizeTitle

File: plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx [hunk 1]
```diff
@@ -223,20 +223,22 @@ export const NotificationsTable = ({
...
-                            onSwitchReadStatus([notification.id], true);
-                          }
-                        }}
-                      >
-                        {notification.payload.title}
-                      </Link>
-                    ) : (
-                      notification.payload.title
-                    )}
+                    <Box sx={{ fontWeight: 'bold' }}>
```
Comment (Rugvip) [inline plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx:226]: is there a Typography variant that you can use instead? thinking h5 or h6 might fit

Comment (mareklibra) [reply plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx]: Those h5 or h6 are still to big, distracting.
Trying `subtitle1`. Please see the later screenshot. Wdyt?

**Comment by mareklibra** [comment]
Trying typography with `subtitle`:

Screenshot from 2024-11-13 09-17-07

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**